### PR TITLE
[CICD] delete old workflow runs

### PR DESCRIPTION
**Pull request by geekflyer**
this keeps the Actions workflow list tidy

### CODE REVIEW

**Review by rustielin — COMMENTED** [review]
The action looks good to me. Surprised that there's no way to do it from the GH UI. I'll leave the final review to max though since it touches the ecosystem prettier

File: .github/ts-tasks/prune-github-workflow-runs.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  core.info(`\nFound the following workflow files in the repo:\n${workflowFilesPresentInRepo.join("\n")}`);
+
+  const workflowResponse = await ghClient.rest.actions.listRepoWorkflows({
+    owner,
+    repo,
+  });
+
+  const obsoleteWorkflows = workflowResponse.data.workflows.filter(
+    (workflow) => !workflowFilesPresentInRepo.includes(path.basename(workflow.path)),
```
Comment (rustielin) [inline .github/ts-tasks/prune-github-workflow-runs.ts:40]: Are workflows always named after their filenames? I thought they were overridden in the yaml spec

**Comment by geekflyer** [comment]
> The action looks good to me. Surprised that there's no way to do it from the GH UI. I'll leave the final review to max though since it touches the ecosystem prettier

you can do it via the UI, but you have to delete every single workflow run individually lol. So lots of clicking.

File: .github/ts-tasks/prune-github-workflow-runs.ts (hunk 1 above)
Comment (geekflyer) [reply .github/ts-tasks/prune-github-workflow-runs.ts:40]: no they are not, but my code uses the workflow.path property as unique identifier (the api also exposes the .path for obsolete workflows).

**Comment by geekflyer** [comment]
> The action looks good to me. Surprised that there's no way to do it from the GH UI. I'll leave the final review to max though since it touches the ecosystem prettier

just had a discussion with @CapCap  offline about the prettier config. I undid part of my change so that it doesn't change the formatting of the SDK. We'll have a seperate discussion between all the JS/TS folks on monday about consistent formatting, but I think this PR should be good to now.

**Pull request merged**